src/Tasks/Unzip.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Threading;
+
+namespace Microsoft.Build.Tasks
+{
+    /// <summary>
+    /// Represents a task that can extract a .zip archive.
+    /// </summary>
+    public sealed class Unzip : TaskExtension, ICancelableTask
+    {
+        /// <summary>
+        /// Stores a <see cref="CancellationTokenSource"/> used for cancellation.
+        /// </summary>
+        private readonly CancellationTokenSource _cancellationToken = new CancellationTokenSource();
+
+        /// <summary>
+        /// Gets or sets a <see cref="ITaskItem"/> with a destination folder path to unzip the files to.
+        /// </summary>
+        [Required]
+        public ITaskItem DestinationFolder { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether read-only files should be overwritten.
+        /// </summary>
+        public bool OverwriteReadOnlyFiles { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether files should be skipped if the destination is unchanged.
+        /// </summary>
+        public bool SkipUnchangedFiles { get; set; } = true;
+
+        /// <summary>
+        /// Gets or sets an array of <see cref="ITaskItem"/> objects containing the paths to .zip archive files to unzip.
+        /// </summary>
+        [Required]
+        public ITaskItem[] SourceFiles { get; set; }
```

Comment:
What happens if there's files in multiple archives with the same path? Should there be an option to warn or error about this?

Reply:
I think its okay to just ignore it, do you want me to log something?